router/http/src/main/scala/io/buoyant/router/http/MethodEnvironmentTenantHostIdentifier.scala

```diff
@@ -2,36 +2,57 @@ package io.buoyant.router.http
 
 import com.twitter.finagle.{Dtab, Path}
 import com.twitter.finagle.buoyant.Dst
-import com.twitter.finagle.http.{Request, Version}
+import com.twitter.finagle.http.{Fields, Request, Version}
 import com.twitter.util.Future
 import io.buoyant.router.RoutingFactory.{IdentifiedRequest, Identifier, RequestIdentification, UnidentifiedRequest}
 
-object MethodTenantHostIdentifier {
+object MethodEnvironmentTenantHostIdentifier {
 
   val TenantHeader = "X-Medallia-Rpc-Tenant"
+  val EnvironmentHeader = "X-Medallia-Rpc-Environment"
+  val ForwardedHeader = "Forwarded"
 
   def mk(
     prefix: Path,
     baseDtab: () => Dtab = () => Dtab.base
-  ): Identifier[Request] = MethodTenantHostIdentifier(prefix, baseDtab, TenantHeader)
+  ): Identifier[Request] = MethodEnvironmentTenantHostIdentifier(prefix, baseDtab, TenantHeader, EnvironmentHeader, ForwardedHeader)
 }
 
-case class MethodTenantHostIdentifier(
+case class MethodEnvironmentTenantHostIdentifier(
   prefix: Path,
   baseDtab: () => Dtab = () => Dtab.base,
-  tenantHeader: String
+  tenantHeader: String,
+  environmentHeader: String,
+  forwardedHeader: String
 ) extends Identifier[Request] {
 
   private[this] def mkPath(path: Path): Dst.Path =
     Dst.Path(prefix ++ path, baseDtab(), Dtab.local)
 
+  private def getForwardedHost(headerMap: Map[String, String]): Option[String] = {
```

Comment:
Let's extract the Forwarded logic to a separate class, preferably a Filter that you can enable via configuration.